components/Calendar/ReactCalendar.tsx

```diff
@@ -0,0 +1,186 @@
+                current = add(nextInterval, { minutes: breakTime });
+            }
+
+            return newTimes;
+        };
+
+        setTimes(getTimes());
+    }, [date.justDate]);
+
+    const handleDateClick = (selectedDate: Date) => {
+        setDate({
+            justDate: selectedDate,
+            dateTime: null
+        });
+        if (isMobile) {
+            setShowCalendar(false);
+        }
+        setShowForm(false);
+    };
+
+    const handleButtonClick = (start: Date) => {
+        setDate((prev) => ({ ...prev, dateTime: start }));
+        setShowForm(true);
+    };
+
+    const handleFormSubmit = (e: React.FormEvent) => {
+        e.preventDefault();
+        setShowForm(false);
+    };
+
+    const handleBackButtonClick = () => {
+        setShowForm(false);
+        setShowCalendar(true);
+    };
+
+    return (
+        <div className="flex items-center justify-center">
+            <div className="flex md:flex-row items-center justify-center rounded-3xl relative w-full md:w-auto p-10 sm:p-0 gap-8">
+                {showCalendar && (
+                    <div className={`md:block ${showForm ? 'hidden' : 'block'} transition-all`}>
+                        <Calendar
+                            minDate={new Date()}
+                            onClickDay={handleDateClick}
+                            view="month"
+                            className="react-calendar p-2"
+                        />
+                    </div>
+                )}
+                {date.justDate && !showForm && (
+                    <div className={`sm:flex flex-col w-full md:w-80 items-center h-calendarHeight rounded-3xl shadow-xl gap-1 ${!showCalendar && date.justDate ? 'flex' : 'hidden'}`}>
+                        <div className="bg-orange mb-1 h-16 w-full rounded-t-3xl flex items-center justify-center text-black">
+                            {format(date.justDate, 'MMMM dd, EEEE')}
+                        </div>
+                        {times.map(({ start, end }) => (
+                            <button
+                                className="text-black p-2 rounded-4xl border border-orange max-w-full w-64 h-8 items-center flex justify-center text-lg"
+                                key={start.toString()}
+                                type="button"
+                                onClick={() => handleButtonClick(start)}
+                            >
+                                {format(start, 'kk:mm')} - {format(end, 'kk:mm')}
+                            </button>
+                        ))}
+                    </div>
+                )}
+                {showForm && (
+                    <div className="flex flex-col w-full md:w-80 items-center h-calendarHeight rounded-3xl shadow-xl">
+                        <div className="bg-orange h-16 w-full rounded-t-3xl flex items-center justify-center text-black">
+                            {format(date.justDate, 'MMMM dd, EEEE')}
```

Comment:
It's not in the design, but I think the selected time from the previous step should be displayed alongside the date.
![image](https://github.com/user-attachments/assets/fa4ff258-7868-4fa1-9259-d41ed6239b26)